Dijkstra Algorithm/README.md

```diff
@@ -0,0 +1,287 @@
+# Dijkstra's algorithm
+
+This [algorithm](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm) was invented in 1956 by Edsger W. Dijkstra. 
+
+It can be used, when you have one source vertex and want to find the shortest paths to all other vertices in the graph.
+
+The best example is road network. If you wnat to find the shortest path from your house to your job, then it is time for the Dijkstra's algorithm.
+
+I have created **VisualizedDijkstra.playground** to improve your understanding of the algorithm's flow. Besides, below is step by step algorithm's description.
+
+Let's imagine, you want to go to the shop. Your house is A vertex and there are 4 possible stores around your house. How to find the closest one/ones? Luckily, you have graph, that connects your house with all these stores. So, you know what to do :)
+
+## Initialization
+
+When the algorithm starts to work initial graph looks like this:
+
+<img src="Images/image1.png" height="250" />
+
+The table below represents graph state:
+
+|                           |  A  |  B  |  C  |  D  |  E  |
+|:------------------------- |:---:|:---:|:---:|:---:|:---:|
+| Visited                   |  F  |  F  |  F  |  F  |  F  |
+| Path Length From Start    | inf | inf | inf | inf | inf |
+| Path Vertices From Start  | [ ] | [ ] | [ ] | [ ] | [ ] |
+
```

Comment:
Before you introduce this table for computing Dijkstra's algorithm, its sometimes helpful to give the reader a concrete goal to think about. Tell them their goal is to find the shortest path from **A** to **E**:

<img width="636" alt="screen shot 2017-05-01 at 5 25 11 pm" src="https://cloud.githubusercontent.com/assets/11384608/25599707/27e885e2-2e93-11e7-9e5d-856bf405419d.png">

Afterwards, you can talk about how to reason about it. That's where the table comes into play.